UI for selecting a Pay Annually software plan

nospame opened the pull request:
## Product Description
Enables selecting a Pay Annually plan instead of directing users to "Talk to Sales" and adds a contact section on the Select Plan page.
Before:
![image](https://github.com/user-attachments/assets/1c0e76be-4d84-4c91-8c3f-7d503433495b)

After:
![image](https://github.com/user-attachments/assets/e89ecb18-550d-441d-897a-c45442b7f452)

## Technical Summary
https://dimagi.atlassian.net/browse/SAAS-17498
Follows #36483
Removes the check that swapped out the "Next" button for "Talk to Sales" when a Pay Annually plan was selected and re-adds a contact form at the bottom of the page. Also makes a couple additions to the form on the backend to ensure `is_annual_plan` is passed all the way through correctly when selecting the plan, confirming, and entering billing info.


## Safety Assurance

### Safety story
Primarily low-stakes UI changes here, and automated testing for the `ConfirmNewSubscriptionForm` should help guard against regression in that area. Tested locally to see that UI looks and functions correctly, will test on staging, and will get QA.

### Automated test coverage
`TestConfirmNewSubscriptionForm`

### QA Plan
Yes, this will get QA before it is released.

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

nospame committed:
Remove oShowNext and its uses

This has the effect of allowing selection of Pay Annually plans

nospame committed:
Current plan -> current edition

nospame committed:
Rename internal property 'selected_annual_plan' and add is_annual_plan form field

nospame committed:
Make is_annual_plan check on view more robust

nospame committed:
Always show description from the selected plan

nospame committed:
Add currentIsAnnualPlan to check if plans are the same

This enables moving to the same edition with the alternate pay annually/pay monthly choice

nospame committed:
Add 'talk to our team' link at bottom of Select Plan page

nospame committed:
Disable next button when Free edition is selected

nospame committed:
Bootstrap 5 Migration - Rebuilt diffs

biyeun commented on corehq/apps/domain/forms.py:
```diff
@@ -2008,7 +2013,7 @@ def send_prepayment_invoice(self, date_start, date_end):
 
     @property
-    def is_annual_plan(self):
+    def selected_annual_plan(self):
```
`is_annual_plan_selected` might be better?

nospame replied:
1fec17a

biyeun commented on corehq/apps/domain/templates/domain/bootstrap5/confirm_plan.html:
```diff
@@ -26,6 +26,7 @@
     >
       {% csrf_token %}
       <input type="hidden" value="{{ plan.edition }}" name="plan_edition" />
+      <input type="hidden" value="{{ is_annual_plan }}" name="is_annual_plan" />
```
might be good to use `BooleanField` in the form and crispy forms if possible...would reduce overall html boilerplate and would have the added benefit of not needing the `lower() == 'true'` check below. not blocking, but should be a followup item.

nospame replied:
ed7bcc3

biyeun commented on corehq/apps/domain/templates/domain/bootstrap3/select_plan.html:
```diff
@@ -280,6 +280,30 @@ <h4 class="text-center">
+          {% blocktrans %}
+            Questions about which software plan is right for you?
+          {% endblocktrans %}
+          <button
```
could be `<a href="#" ....>`?

biyeun reviewed (approved):
nothing blocking in comments, but definitely some nice to-dos in there

biyeun commented on corehq/apps/accounting/static/accounting/js/pricing_table.js:
```diff
@@ -23,28 +24,30 @@ var PricingTable = function (options) {
     self.invoicingContact = options.invoicingContact;
 
-    self.oSelectedPlan = ko.observable(options.currentPlan);
+    self.oSelectedPlan = ko.observable(options.currentEdition);
```
rename to `oSelectedEdition`

nospame replied:
https://github.com/dimagi/commcare-hq/pull/36499/commits/8834204887f6d555f4e87d1df0e634bf8bf31304

nospame committed:
oSelectedPlan and oCurrentPlan to 'Edition'

nospame committed:
selected_annual_plan property -> is_annual_plan_selected method

No strict need to be a property, and it's more consistent with other checks if it's not.

nospame committed:
Fix self.is_annual_plan checks

nospame committed:
Make 'talk to our team' just a link because we're not using the form data anyway

nospame committed:
Add general plan question form

nospame committed:
Bootstrap 5 Migration - Rebuilt diffs

biyeun commented on corehq/apps/domain/views/accounting.py:
```diff
@@ -1318,6 +1318,22 @@ def back_button(self):
 
+class GeneralPlanQuestionView(ContactFormViewBase):
+    urlname = 'select_plan_general_question'
+    request_type = 'Plan Question'
```
i think this is accounts-facing text, right? not user-facing?

biyeun reviewed (approved):
looks good. i had one comment on the request_type text, but we already follow that format in other places so it's not blocking.

nospame committed:
Don't allow continue if selected plan is enterprise

nospame committed:
Remove unused 'keep subscription' email

gherceg reviewed (approved):
I just reviewed the last two commits and they look good to me.

mjriley commented on corehq/apps/accounting/static/accounting/js/pricing_table.js:
```diff
@@ -40,10 +40,11 @@ var PricingTable = function (options) {
     self.oShowAnnualPricing = ko.observable(options.currentIsAnnualPlan);
 
     self.oIsSubmitDisabled = ko.computed(function () {
+        var isSubscribablePlan = !!self.oSelectedEdition() && !['free', 'enterprise'].includes(self.oSelectedEdition());
```
very small nit, but if we're introducing new variables, I think we should try to use modern `let` or `const` rather than `var`

nospame replied:
I agree, but I was more interested in consistency with the surrounding code here. In an ideal world I'd find the time to update this whole file to use `let` and `const` as appropriate.

mjriley reviewed (approved):
Looks good
